middleware/errors/errors.go

```diff
@@ -36,14 +37,17 @@ func (h ErrorHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, er
 			// Write error to response instead of to log
 			w.Header().Set("Content-Type", "text/plain; charset=utf-8")
 			w.WriteHeader(status)
-			fmt.Fprintln(w, errMsg)
+			fmt.Fprintln(responseRecorder, errMsg)
 			return 0, err // returning < 400 signals that a response has been written
 		}
 		h.Log.Println(errMsg)
 	}
 
 	if status >= 400 {
-		h.errorPage(w, r, status)
+		// Check there is no response body already sent
+		if responseRecorder.Size() == 0 {
+			h.errorPage(responseRecorder, r, status)
+		}
```

Comment:
Now that your PR has been merged, we can just check the content-length header like you do below; no need to make a ResponseRecorder. :+1: